stablehlo/dialect/Base.cpp

```diff
@@ -300,6 +300,11 @@ FailureOr<TensorType> inferTypeWithCustomFn(
         Optional<Location>, int64_t, int64_t, int64_t, int64_t, int64_t)>
         inferDimAndBoundFn) {
   auto rank = rankedTypes[0].getRank();
+  for (auto& type : rankedTypes) {
+    if (type.getRank() != rank) {
+      return failure();
+    }
+  }
```

Comment:
Thanks!
Can we use the loop below to add the check and on failure, return with a diagnostic message. Something like (feel free to use other messages)
```
  return emitOptionalError(location, "Mismatched ranks of types", rankedTypes[0].getRank(), " vs ", rankedTypes[i].getRank());
```